chia/_tests/core/data_layer/test_data_rpc.py

```diff
@@ -3431,3 +3433,204 @@ async def test_unsubmitted_batch_update(
+                assert stderr == b"" or b"_ProactorBasePipeTransport.__del__" in stderr
+            if submit_on_chain:
+                update_tx_rec0 = bytes32.from_hexstr(res["tx_id"][0])
+            else:
+                assert res == {"success": True}
+        elif layer == InterfaceLayer.client:
+            client = await DataLayerRpcClient.create(
+                self_hostname=self_hostname,
+                port=rpc_port,
+                root_path=bt.root_path,
+                net_config=bt.config,
+            )
+            try:
+                res = await client.update_multiple_stores(
+                    changelist=changelist,
+                    submit_on_chain=submit_on_chain,
+                    fee=None,
+                )
+            finally:
+                client.close()
+                await client.await_closed()
+            if submit_on_chain:
+                update_tx_rec0 = bytes32.from_hexstr(res["tx_id"][0])
+            else:
+                assert res == {"success": True}
+        else:  # pragma: no cover
+            assert False, "unhandled parametrization"
+
+        if not submit_on_chain:
+            if layer == InterfaceLayer.direct:
+                res = await data_rpc_api.submit_all_pending_roots({})
+                update_tx_rec0 = res["tx_id"][0]
+            elif layer == InterfaceLayer.funcs:
+                res = await submit_all_pending_roots_cmd(
+                    rpc_port=rpc_port,
+                    fee=None,
+                    fingerprint=None,
+                    root_path=bt.root_path,
+                )
+                update_tx_rec0 = bytes32.from_hexstr(res["tx_id"][0])
+            elif layer == InterfaceLayer.cli:
+                args = [
+                    sys.executable,
+                    "-m",
+                    "chia",
+                    "data",
+                    "submit_all_pending_roots",
+                    "--data-rpc-port",
+                    str(rpc_port),
+                ]
+                process = await asyncio.create_subprocess_exec(
+                    *args,
+                    env={**os.environ, "CHIA_ROOT": str(bt.root_path)},
+                    stdout=asyncio.subprocess.PIPE,
+                    stderr=asyncio.subprocess.PIPE,
+                )
+                await process.wait()
+                assert process.stdout is not None
+                assert process.stderr is not None
+                stdout = await process.stdout.read()
+                res = json.loads(stdout)
+                stderr = await process.stderr.read()
+                assert process.returncode == 0
+                if sys.version_info >= (3, 10, 6):
+                    assert stderr == b""
+                else:  # pragma: no cover
+                    # https://github.com/python/cpython/issues/92841
+                    assert stderr == b"" or b"_ProactorBasePipeTransport.__del__" in stderr
```

Comment:
Probably worth a helper function to avoid duplicating this boilerplate?  Even if it is just a very local helper and not something trying to be super general.